### DISCUSSION TITLE
How can we catch the `Error: Invalid signature` exception within MultiSamlStrategy?

### DISCUSSION THREAD

**Discussion by jeff-pal** [bergie/passport-saml#801] [Q&A] [answered]
If we have an invalid certificate in `done(null, provider.configuration)` we get a `Invalid signature` exception. I'd expect to catch that exception but I'm not sure where it does throw the exception. So how can we handle that error?

```javascript
passport.use(
  new MultiSamlStrategy(
    {
      passReqToCallback: true, // makes req available in callback
      getSamlOptions: function (request, done) {
        findProvider(request, function (err, provider) {
          try {
            if (err) {
              return done(err);
            }
            return done(null, provider.configuration); // <------------- The error source
          } catch (e) {
            console.error(e.message); //  <------------- I'd expect to catch the error here
          }
        });
      },
    },
    function (req, profile, done) {
      // for signon
      findByEmail(profile.email, function (err, user) {
        if (err) {
          return done(err);
        }
        return done(null, user);
      });
    },
  )
);
```

**Comment by cjbarth** [comment, marked as answer]
You should be getting an Passport error, so you would handle it that way. Perhaps [this](https://stackoverflow.com/questions/15711127/express-passport-node-js-error-handling) helps. See [this](https://www.passportjs.org/concepts/authentication/strategies/#:~:text=cb(null%2C%20false)%3B-,If%20an%20error%20occurs,return%20cb(err)%3B,-It%20is%20important) too. This will typically be handled by Express and in the [typical Node-back style](https://nodejs.org/en/knowledge/errors/what-are-the-error-conventions/).
